src/main.rs

```diff
@@ -73,6 +85,13 @@ fn main() -> Result<()> {
     let config = Config::get_or_create()?;
     let _ = fs::create_dir(&config.temporary_project_dir);
 
+    #[cfg(feature = "generate")]
+    match cli.subcommand {
+        Some(Subcommand::Generate(args)) => args.generate(config)?,
+        None => execute(cli, config)?,
+    }
+
+    #[cfg(not(feature = "generate"))]
     execute(cli, config)?;
```

Comment:
There is probably a more compact way to write this but I like the explicitness here, it's easy to read :+1: